bsp/thead-smart/drivers/board.c

```diff
@@ -0,0 +1,70 @@
+/*
+ * File      : board.c
+ * This file is part of RT-Thread RTOS
+ * COPYRIGHT (C) 2018, RT-Thread Development Team
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Change Logs:
+ * Date           Author       Notes
+ * 2017-08-10     Urey         first implementation
+ */
+
+#include <rthw.h>
+#include <rtthread.h>
+
+#include "board.h"
+#include "csi_core.h"
+#include "drv_usart.h"
+
+extern int __bss_end__;
+#define SYS_HEAP_BEGIN    (&__bss_end__)
+
+#include "core_rv32.h"
+
+
+extern usart_handle_t console_handle;
+extern void ioreuse_initial(void);
+
+
+/**
+ * This function will initial smartl-evb(e906) board.
+ */
+void rt_hw_board_init(void)
+{
+/* usart init ? */
+
+//    phobos_ioreuse_initial();
```

Comment:
Please remove unnecessary code.

Reply:
When the functions rt_hw_board_init() and rt_show_version() are executed, the global interrupt has been turned on. The clock tick interrupt processing function calls rt_tick_increase(), and the call of the rt_tick_increase() function is executed after rt_system_timer_init(), which will cause the use of uninitialized pointer rt_timer_list->next And rt_timer_list->nextprev, 

This will cause a runtime error！